src/Emulator/Visuals/Visuals.js

```diff
@@ -1,11 +1,236 @@
-import React from 'react';
+import React, {useRef, useState, useEffect} from 'react';
 import './Visuals.css';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSearchPlus, faSearchMinus, faExpand, faEyeSlash} from '@fortawesome/free-solid-svg-icons';
+import Draggable from 'react-draggable';
+import { parse } from 'path';
 
 const Visuals = () => {
+  const containerHeight = '50vh';
+  const containerWidth = '75vw';
+  const additionalImageSpace = 20;
+
+  const [zoom, setZoom] = useState(true);
+  const [hideImage, setHideImage] = useState(false);
+  const [toggleMax, setMax] = useState(false);
+  const imageRef = useRef();
+  const [containerDimensions, setContainerDimensions] = useState({height: containerHeight, width: containerWidth});
+  const [contentDimensions, setContentDimensions] = useState({height: 0, width: 0});
+  const [image, setImage] = useState('null');
+  const [, setSrc] = useState('');
+  const [, setDrags] = useState(0);
+  const defaultX = 200;
+  const defaultY = 200;
+  const [pos, setPos] = useState({x: defaultX, y: defaultY});
+  const [handleDetect, setHandleDetect] = useState({x: '', y: ''})
+  const [elementBounds, setElementBounds] = useState({top: contentDimensions.height || 125, bottom: contentDimensions.height|| 125, left: '', right: contentDimensions.width || 125})
```

Comment:
New line per property:
```
{
    top: contentDimensions.height || 125,
    bottom: contentDimensions.height|| 125,
    left: '',
    right: contentDimensions.width || 125
}
```

Why is left a string while the others are 125? Why not 125? What is special about 125? Can it be a proportion of the width, else can you make them a constant at the top?

Reply:
When the element is initially rendered, it's skin-tight to the element so you can't drag by clicking near the element. This just allows the initial render to have some extra room to click and drag around the image. As new images are loaded in, it runs a function that will then automatically add extra space so that you can always drag by clicking near the element no matter the size of the image.